outputs.tf

```diff
@@ -1,10 +1,29 @@
 output "groups" {
-  description = "A nested map of group resources, with group name as the key, and each value map containing the groups's ARN and Unique ID."
-  value = {
-    for group, properties in aws_iam_group.this : group => {
-      arn : properties.arn
-      name : properties.name
-      unique_id : properties.unique_id
+  description = "List of group objects with keys 'arn', 'name' and 'unique_id'"
+  value = concat([for group, properties in aws_iam_group.this : {
```

Comment:
We want outputs to be maps keyed on the resource name:
`module.iam.groups = { group1: { name: ..., arn: ... }, group2: {...}`

Same for users.